TestArchiverAbortEarlyOnError fails on armhf

<!--

Welcome! - We kindly ask that you:

  1. Fill out the issue template below - not doing so needs a good reason.
  2. Use the forum if you have a question rather than a bug or feature request.

The forum is at: https://forum.restic.net

NOTE: Not filling out the issue template needs a good reason, as otherwise it
may take a lot longer to find the problem, not to mention it can take up a lot
more time which can otherwise be spent on development. Please also take the
time to help us debug the issue by collecting relevant information, even if
it doesn't seem to be relevant to you. Thanks!

The forum is a better place for questions about restic or general suggestions
and topics, e.g. usage or documentation questions! This issue tracker is mainly
for tracking bugs and feature requests directly relating to the development of
the software itself, rather than the project.

Thanks for understanding, and for contributing to the project!

-->


Output of `restic version`
--------------------------
restic 0.12.1 compiled with go1.16.5 on linux/amd64


How did you run restic exactly?
-------------------------------
`RESTIC_TEST_FUSE=0 make test`

<!--
This section should include at least:

 * The complete command line and any environment variables you used to
   configure restic's backend access. Make sure to replace sensitive values!

 * The output of the commands, what restic prints gives may give us much
   information to diagnose the problem!
-->

What backend/server/service did you use to store the repository?
----------------------------------------------------------------
Whatever the testsuite uses


Expected behavior
-----------------
The test passes
<!--
Describe what you'd like restic to do differently.
-->

Actual behavior
---------------
```
go test ./cmd/... ./internal/...
go: downloading github.com/google/go-cmp v0.5.6
ok  	github.com/restic/restic/cmd/restic	12.482s
--- FAIL: TestArchiverAbortEarlyOnError (0.08s)
    --- FAIL: TestArchiverAbortEarlyOnError/#01 (0.02s)
        testing.go:30: using low-security KDF parameters for test
        config.go:65: disabling check of the chunker polynomial
        archiver_test.go:2020: chdir to /tmp/restic-test-3497978820
        archiver_test.go:2048: Snapshot return error: test error
        archiver_test.go:2050: track fs: map[.:1 dir:1 dir/file1:1 dir/file2:1 dir/file3:1 dir/file4:1 dir/file5:1 dir/file6:1 dir/file7:1]
        archiver_test.go:2054: opened dir/file7 1 times, want 0
        archiver_test.go:2057: chdir back to /home/buildozer/aports/community/restic/src/restic-0.12.1/internal/archiver
FAIL
FAIL	github.com/restic/restic/internal/archiver	5.981s
```
<!--
In this section, please try to concentrate on observations, so only describe
what you observed directly.
-->

Steps to reproduce the behavior
-------------------------------
Try to run restic tests on armhf

<!--
The more time you spend describing an easy way to reproduce the behavior (if
this is possible), the easier it is for the project developers to fix it!
-->

Do you have any idea what may have caused this?
-----------------------------------------------
No


Do you have an idea how to solve the issue?
-------------------------------------------
I can provide more in-depth logs from the failed builder if the test provides

Confirmed with QEMU on a Linux/amd64 host, Go 1.17, bf9c8771a469f1ee0d4b6fab78791d36426371b2:

```
$ GOARCH=arm go test -run=TestArchiverAbortEarlyOnError/#01 -count=100
--- FAIL: TestArchiverAbortEarlyOnError (0.27s)
    --- FAIL: TestArchiverAbortEarlyOnError/#01 (0.27s)
        testing.go:30: using low-security KDF parameters for test
        config.go:65: disabling check of the chunker polynomial
        archiver_test.go:2020: chdir to /tmp/restic-test-1422233881
        archiver_test.go:2048: Snapshot return error: test error
        archiver_test.go:2050: track fs: map[.:1 dir:1 dir/file1:1 dir/file2:1 dir/file3:1 dir/file4:1 dir/file5:1 dir/file6:1 dir/file7:1]
        archiver_test.go:2054: opened dir/file7 1 times, want 0
        archiver_test.go:2057: chdir back to /home/me/src/restic/restic/internal/archiver
FAIL
exit status 1
FAIL	github.com/restic/restic/internal/archiver	10.697s
```

I couldn't trigger this without GOARCH=arm, but neither can I trigger it consistently with that GOARCH. May still be a concurrency bug rather than anything architecture-specific.

--- a/internal/archiver/archiver_test.go
+++ b/internal/archiver/archiver_test.go
@@ -1998,7 +1998,6 @@ func TestArchiverAbortEarlyOnError(t *testing.T) {
 				filepath.FromSlash("dir/file2"): 1,
 				filepath.FromSlash("dir/file3"): 1,
 				filepath.FromSlash("dir/file4"): 1,
-				filepath.FromSlash("dir/file7"): 0,
 				filepath.FromSlash("dir/file8"): 0,
 				filepath.FromSlash("dir/file9"): 0,
 			},

